<a href="https://colab.research.google.com/github/dpetrovic89/Telenor-Segments/blob/main/What_features_can_describe_each_segment%3F.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Get Data

In [ ]:
import pandas as pd

In [ ]:
pd.set_option('display.max_colwidth', None)

In [ ]:
pd.set_option('display.max_columns', 500)

In [ ]:
df_description = pd.read_csv('https://raw.githubusercontent.com/dpetrovic89/data/main/Variable-Explanation-Test-Data-Scientist_2022.csv', header=None, )

In [ ]:
df_description.head(15)

0  ...   2
0          SUBSCRIBER_ID  ... NaN
1               BNUM_OUT  ... NaN
2                BNUM_IN  ... NaN
3                  CHURN  ... NaN
4         PROD_CNT_MACRO  ... NaN
5            REV_BUN_MAC  ... NaN
6                REV_OUT  ... NaN
7      USAGE_OUT_INT_DUR  ... NaN
8   USAGE_OUT_OFFNET_DUR  ... NaN
9    USAGE_OUT_ONNET_DUR  ... NaN
10            LNE_TENURE  ... NaN
11             TOPUP_AMT  ... NaN
12             TOPUP_CNT  ... NaN
13               Segment  ... NaN

[14 rows x 3 columns]

In [ ]:
df= pd.read_csv('https://raw.githubusercontent.com/dpetrovic89/data/main/Data-Churn-Test-Data-Scientist_2021.csv' )

In [ ]:
df.head()

SUBSCRIBER_ID  BNUM_OUT  BNUM_IN  ...  TOPUP_AMT  TOPUP_CNT    Segment
0              1       2.0      3.0  ...        NaN        NaN  Segment 1
1              2      52.0     32.0  ...        2.0        3.0  Segment 2
2              3       NaN      NaN  ...        NaN        0.0  Segment 3
3              4       4.0      5.0  ...        4.0        2.0  Segment 1
4              5       0.0     14.0  ...        6.0        2.0  Segment 5

[5 rows x 14 columns]

# Interpret Data

In [ ]:
df['Segment'].describe()

count          4998
unique            4
top       Segment 5
freq           1647
Name: Segment, dtype: object

In [ ]:
df.groupby('Segment').describe()

SUBSCRIBER_ID                                                  \
                  count         mean          std  min      25%     50%   
Segment                                                                   
Segment 1        1081.0  2600.608696  1447.757347  1.0  1370.00  2654.0   
Segment 2        1410.0  2457.969504  1445.124857  2.0  1223.25  2415.5   
Segment 3         860.0  2400.654651  1451.436446  3.0  1100.25  2363.0   
Segment 5        1647.0  2522.074681  1431.724118  5.0  1299.50  2552.0   

                           BNUM_OUT                                         \
               75%     max    count       mean        std  min   25%   50%   
Segment                                                                      
Segment 1  3888.00  5000.0   1017.0   4.125860   8.122578  0.0   0.0   1.0   
Segment 2  3747.75  4997.0   1403.0  22.056308  17.319284  0.0  11.0  17.0   
Segment 3  3667.75  4999.0    721.0   4.922330   6.159867  0.0   0.0   3.0   
Segment 5  3701.00  4998.0   1369.0   4.102995   4.625008  0.0   1.0   3.0   

                       BNUM_IN                                               \
            75%    max   count       mean        std  min   25%   50%   75%   
Segment                                                                       
Segment 1   5.0  106.0  1017.0   5.505408   7.426823  0.0   1.0   3.0   7.0   
Segment 2  28.0  158.0  1403.0  26.771205  23.381650  0.0  13.0  22.0  34.0   
Segment 3   8.0   61.0   721.0  11.642164  12.074173  0.0   4.0   9.0  16.0   
Segment 5   6.0   43.0  1369.0  10.211103  13.160225  0.0   3.0   7.0  13.0   

                   CHURN                                               \
             max   count      mean       std  min  25%  50%  75%  max   
Segment                                                                 
Segment 1   84.0  1081.0  1.000000  0.000000  1.0  1.0  1.0  1.0  1.0   
Segment 2  445.0  1410.0  0.002837  0.053206  0.0  0.0  0.0  0.0  1.0   
Segment 3  121.0   860.0  0.000000  0.000000  0.0  0.0  0.0  0.0  0.0   
Segment 5  266.0  1647.0  0.000000  0.000000  0.0  0.0  0.0  0.0  0.0   

          PROD_CNT_MACRO                                                \
                   count      mean       std  min  25%  50%  75%   max   
Segment                                                                  
Segment 1          402.0  0.437811  0.905711  0.0  0.0  0.0  1.0   7.0   
Segment 2          865.0  2.337572  1.852622  0.0  1.0  3.0  4.0  18.0   
Segment 3          378.0  0.412698  0.690249  0.0  0.0  0.0  1.0   3.0   
Segment 5          528.0  0.325758  0.654817  0.0  0.0  0.0  0.0   4.0   

          REV_BUN_MAC                                               REV_OUT  \
                count      mean       std  min  25%  50%  75%   max   count   
Segment                                                                       
Segment 1       112.0  1.875000  1.605873  1.0  1.0  1.0  2.0   9.0  1081.0   
Segment 2       669.0  4.683109  3.606967  1.0  2.0  4.0  6.0  28.0  1407.0   
Segment 3       115.0  1.678261  1.072363  1.0  1.0  1.0  2.0   6.0   685.0   
Segment 5       126.0  1.579365  1.045778  1.0  1.0  1.0  2.0   7.0  1317.0   

                                                        USAGE_OUT_INT_DUR  \
               mean       std  min  25%  50%  75%   max             count   
Segment                                                                     
Segment 1  1.359852  3.502250  0.0  0.0  0.0  1.0  43.0            1081.0   
Segment 2  5.004975  7.157692  0.0  1.0  3.0  6.0  67.0            1407.0   
Segment 3  1.048175  1.873722  0.0  0.0  0.0  1.0  24.0             727.0   
Segment 5  1.350797  2.506516  0.0  0.0  1.0  2.0  39.0            1382.0   

                                                           \
               mean        std  min  25%  50%  75%    max   
Segment                                                     
Segment 1  0.473636   2.653352  0.0  0.0  0.0  0.0   36.0   
Segment 2  2.0

What features can describe each segment?

**Segment 1**: Total Users 1081, Number of different numbers who subscriber calls on average 4.12, Number of different numbers that make calls to subscriber on average per segment 5.50, All users Churn in segment 1, Number of bundles that customer activated (average in last 3 months) on average 0.44, Revenue from bundles that customer activated (average in last 3 months) 1.87, Total Revenue (average in last 3 months) per this segment 1.36, Number of outgoing international minutes (average in last 3 months)  0.47, Number of outgoing minutes to competitor network (average in last 3 months) 3.8, Number of outgoing minutes in the network (average in last 3 months) 19.94, User tenure in months 118, Total topup (refill) amount (average in last 3 months) 336 users generated 5.71 on average, Number of topups (refill) (average in last 3 months) 1.32 

**Segment 2**: Total Users 1410, Number of different numbers who subscriber calls 22.05, Number of different numbers that make calls to subscriber on average per segment 26.77, Few Churn in segment 2, Number of bundles that customer activated (average in last 3 months) on average 2.34, Revenue from bundles that customer activated (average in last 3 months) 4.68, Total Revenue (average in last 3 months) per this segment  5.00, Number of outgoing international minutes (average in last 3 months) 2.03, Number of outgoing minutes to competitor network (average in last 3 months) 16.40, Number of outgoing minutes in the network (average in last 3 months) 223.41, User tenure in months 122, Total topup (refill) amount (average in last 3 months) 1305 users generated 9.17 on average, Number of topups (refill) (average in last 3 months) 3.90 

**Segment 3**: Total Users 860, Number of different numbers who subscriber calls 4.92, Number of different numbers that make calls to subscriber on average per segment 11.64, Zero Churn in segment 3, Number of bundles that customer activated (average in last 3 months) on average 0.41, Revenue from bundles that customer activated (average in last 3 months) 1.68, Total Revenue (average in last 3 months) per this segment 1.04, Number of outgoing international minutes (average in last 3 months) 0.16, Number of outgoing minutes to competitor network (average in last 3 months) 2.8, Number of outgoing minutes in the network (average in last 3 months) 21.25, User tenure in months 111, Total topup (refill) amount (average in last 3 months) 378 generated 3.04 on average, Number of topups (refill) (average in last 3 months) 1.27 

**Segment 5**: Total Users 1647, Number of different numbers who subscriber calls 4.10, Number of different numbers that make calls to subscriber on average per segment 10.21,  Zero Churn in segment 5, Number of bundles that customer activated (average in last 3 months) on average 0.32, Revenue from bundles that customer activated (average in last 3 months) 1.58, Total Revenue (average in last 3 months) per this segment 1.35, Number of outgoing international minutes (average in last 3 months) 0.39, Number of outgoing minutes to competitor network (average in last 3 months) 4.01, Number of outgoing minutes in the network (average in last 3 months) 11.04, User tenure in months 125,  Total topup (refill) amount (average in last 3 months) 716 users generated 3.68 on average, Number of topups (refill) (average in last 3 months) 1.24

**Segment 1**: 

All users  are churned. 

They also get least calls to them 5.5 on average. 

Total refill amount is 2x bigger than segments 3 and 5 and they use same number of refills as segments 3 and 5.

**Segment 2**: 

Most Active and Profitable segment. 

They call on average 4x more than other segments, and they get calls to them 2x to 5x times more. 

Very few churn in segment 2. 

Number of bundles activated is 5x more than rest. 

Revenue from bundles is 3x more than others. 

Total Revenue for this segment is 3x to 5x bigger than others. 

Number of outgoing international minutes are more than 1.5 than others. 

Outgoing minutes to competitor is 4x to 5x than others. 

Outgoing minutes in network is 10x to 20x more than others. 

Total refill amount is 2x to 3x bigger than others and they refill 3x more than rest.

**Segments 3 and 5 are similar**: 

They are active users (Not Churned). 

Difference between them is segment 5 have 3x more outgoing international minutes. 

Segment 5 also have 30% more outgoing minutes to competitor network. 

But segment 3 have 2x more outgoing minutes in the network. 

Total revenue is 30% bigger in segment 5 than in segment 3. 


